File: python/1207.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        arr = sorted(arr)
        count = 1
        counts = []
        prev = arr[0]
        for i in range(1, len(arr)):
            if(prev == arr[i]):
                count+=1
            else:
                counts.append(count)
                prev = arr[i]
                count =1
        counts.append(count)

        counts = sorted(counts)
        prev = counts[0]
        for i in range(1, len(counts)):
            if(counts[i-1] == counts[i]):
                return False
        return True
        

File: cpp/1207.cpp
class Solution {
public:

    bool uniqueOccurrences(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int count=1;
        int prev = arr[0];
        vector<int> counts;
        for(int i=1; i<arr.size(); i++) {
            auto& a = arr[i];
            if(prev==a)
                count++;
            else {
                counts.push_back(count);
                count=1;
                prev = a;
            }
        }
        counts.push_back(count);

        sort(counts.begin(), counts.end());
        prev = counts[0];
        cout<<counts.size();
        for(int i=1; i<counts.size(); i++) {

            if(counts[i]==prev) 
                return false;
            prev = counts[i];
        }
        return true;
    }
};